pkg/sqs/sqs.go

```diff
@@ -98,36 +98,37 @@ func (mh *MessageHandler) handleMessage(msg *sqs.ReceiveMessageOutput) {
 	for m := range msg.Messages {
 		log.Debugf("Message body: %s", string(*msg.Messages[m].Body))
 
-		var message util.Message
-		err := json.Unmarshal([]byte(*msg.Messages[m].Body), &message)
-
 		messageHandle := msg.Messages[m].ReceiptHandle
 
-		if err != nil {
-			log.Error(err)
-			mh.deleteConsumedMessage(messageHandle)
-			return
-		}
+		var messageASG util.ASGMessage
+		var messageSpot util.SpotMessage
+		json.Unmarshal([]byte(*msg.Messages[m].Body), &messageASG)
```

Comment:
You are swallowing the error here. Does the unmarshalling error, if the data does not match the struct? Because I would guess, that it would not return an error. Therefore error checking should work.

Reply:
the thing is that there will always be at least one error. so that is expected and error handling happens implicitly when checking for the keys ins the following lines. not sure what to do about an error other then to log it. and even then it's misleading because that error is expected. :thinking:

Reply:
but let me check the error values again... just to be sure...